pkg/skaffold/config/util.go

```diff
@@ -223,3 +227,34 @@ func IsUpdateCheckEnabled(configfile string) bool {
 	}
 	return cfg == nil || cfg.UpdateCheck == nil || *cfg.UpdateCheck
 }
+
+func ShouldDisplayPrompt(configfile string) bool {
+	if cfg, disabled := isSurveyPromptDisabled(configfile); !disabled {
+		return ifNotSurveyTakenOrPromptNotDisplayed(cfg)
+	}
+	return false
+}
+
+func isSurveyPromptDisabled(configfile string) (*ContextConfig, bool) {
+	cfg, err := GetConfigForCurrentKubectx(configfile)
+	if err != nil {
+		return nil, false
+	}
+	return cfg, cfg != nil && cfg.Survey != nil && *cfg.Survey.DisablePrompt
+}
+
+func ifNotSurveyTakenOrPromptNotDisplayed(cfg *ContextConfig) bool {
```

Comment:
nit: rename to `recentlyPromptedOrTaken` and negate the return logic to reflect it 

(negation in makes things less readable - technically the name should be this way `notTakenAndNotPromptedRecently` not or...)